app/models/tagged_news.rb

```diff
@@ -1,12 +1,26 @@
-class TaggedNews
+class TaggedNews < SimpleDelegator
   def self.all(article)
     news_items = Mas::Cms::News.all(
       params: {
         document_type: [Mas::Cms::News::PAGE_TYPE],
         tag: Array(article.tags)
       }
-    )
+    ).reject { |news_item| news_item.slug == article.slug }
 
-    news_items.reject { |news_item| news_item.slug == article.slug }
+    sort_by_published_date(news_items)
+  end
+
+  def self.sort_by_published_date(news_items)
+    news_items.map do |news_item|
+      new(news_item)
+    end.sort_by(&:published_date).reverse
```

Comment:
I don't like the idea of sorting the collection so late, it should be a parameter to send to the `.all` method in the `Mas::Cms::Resource` – plus, it cannot be re-used.

But I guess for the purpose of this PR it's a good solution.

Reply:
It is still a code smell to me that the `all` method is defining a sort order. I agree with @your that it should be a parameter passed to the method. As a compromise, what if it was set to a default value which at least would be more declarative that the sort order is being defined, wdyt?

Reply:
> I agree with @your that it should be a parameter passed to the method. As a compromise, what if it was set to a default value which at least would be more declarative that the sort order is being defined, wdyt?

I can't see it through yet. Something like these?

```ruby
  def self.all(article, order_by: :published_date)
    # ...
  end
```

Reply:
@tomas-stefano yes something like
```
def self.all(article, order_by: :published_date)
  # ...
end
```

Reply:
@margOnline I need to pair with you because I still don't know how to implement your suggestions. Can we catchup tomorrow, please?

Reply:
I created [this branch](https://github.com/moneyadviceservice/fin_cap/tree/refactor/sort-news-by-custom-date) to give you an idea of what I meant. _(Tests haven't been updated)_

Reply:
I understand now. Thank you for the clarification.

The problem is that `TaggedNews.all(documents).sort_by_custom_order_date(documents)` does not work because `TaggedNews.all(documents)` will return an Array of Mas::Cms::News so the method `order_by_published_date` would not be available.

I added the sort inside the .all because it is being used in a lot of controllers:

```
app/controllers/homepages_controller.rb
7:    @latest_news = TaggedNews.all(@homepage).map do |news|

app/controllers/lifestages_controller.rb
10:    @latest_news = TaggedNews.all(lifestage).map do |news_item|

app/controllers/regional_strategies_controller.rb
5:    @latest_news = TaggedNews.all(@regional_strategy).map do |news|

app/controllers/thematic_reviews_controller.rb
18:    @latest_news = TaggedNews.all(@thematic_review).map do |news_item|

app/controllers/uk_strategies_controller.rb
5:    @latest_news = TaggedNews.all(@uk_strategy).map do |news|
```

Otherwise I will need to add the repetitive sort by method on all controllers:
```
 @latest_news = TaggedNews.all(@homepage).map do |news|
    NewsTemplate.new(news)
 end.sort_by(&:published_date).reverse
```

But one thing that I like from your solution is to move the method to mas cms news object. Let's catchup in slack about this, please?

Reply:
I just move the published date for individual news to the mas cms news. Where it should be. The other aspect of the sort we need to talk about it. ♻️